calliope/backend/checks.py

```diff
@@ -61,6 +61,11 @@ def _is_in(loc_tech, set_or_var):
             return False
 
     def _set_inf_and_warn(loc_tech, var, warnings, warning_text):
+        if model_data[var].dtype == int:
```

Comment:
I don't think this is a good idea.  I suspect it will fail when the dataset is created and used on different platforms: 32-bit vs 64-bit.  If your intention was to check for an integer in a platform agnostic manner, try the following:
```suggestion
        if model_data[var].dtype.kind == "i":
```
see: [`numpy.dtype` docs](https://numpy.org/doc/1.18/reference/generated/numpy.dtype.kind.html#numpy.dtype.kind).

That said, maybe it is just simplest to unilaterally call `astype(float)`?  But if I haven't quite grasped the larger context, please ignore this second comment.

Reply:
Yeah, possibly just setting as float without checking would be the least likely to cause issues